src/main/scala/mesosphere/marathon/storage/migration/legacy/MigrationTo_1_4_9.scala

```diff
@@ -0,0 +1,60 @@
+package mesosphere.marathon.storage.migration.legacy
+
+import akka.Done
+import akka.stream.Materializer
+import akka.stream.scaladsl.{ Flow, Keep, Sink }
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.state._
+import mesosphere.marathon.storage.migration.legacy.MigrationTo_1_4_6.Environment
+import mesosphere.marathon.storage.repository.InstanceRepository
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+@SuppressWarnings(Array("ClassNames"))
+class MigrationTo149(instanceRepository: InstanceRepository)(implicit ctx: ExecutionContext, mat: Materializer) extends StrictLogging {
+
+  def migrate(): Future[Done] = {
+    MigrationTo149.migrateUnreachableInstances(instanceRepository)(Environment(sys.env), ctx, mat)
+  }
+}
+
+object MigrationTo149 extends StrictLogging {
+
+  private def changeUnreachableStrategyForInstances(instance: Instance): Instance = {
+    instance.copy(unreachableStrategy = MigrationTo_1_4_6.changeUnreachableStrategy(instance.unreachableStrategy))
+  }
+
+  def migrateUnreachableInstances(instanceRepository: InstanceRepository)(implicit env: Environment, ctx: ExecutionContext, mat: Materializer): Future[Done] = {
+    logger.info("Starting unreachable strategy migration to 1.4.9")
+
+    val instanceSink =
+      Flow[Instance]
+        .mapAsync(Int.MaxValue)(instanceRepository.store)
```

Comment:
In my view `.mapAsync(Int.MaxValue)` looks like we don't want to have any backpressure :) Do you think it makes sense to change `Int.MaxValue` to a lower value?

Reply:
Migrating discussion to https://github.com/mesosphere/marathon/pull/5659#issuecomment-338158484